src/main/kotlin/com/manimdsl/frontend/AbstractSyntaxTreeVisitor.kt

```diff
@@ -0,0 +1,50 @@
+package com.manimdsl.frontend
+
+interface AbstractSyntaxTreeVisitor<T> {
+
+    /** All statements making up program **/
+    fun visitProgramNode(programNode: ProgramNode): T?
```

Comment:
Is there a reason the visitor can return something that is nullable

Reply:
@arj119 Can we not delete this?

Reply:
Where do we use this?

Reply:
It's up to our design, but it means we aren't necessarily tied to returning a T if there is a case where we shouldn't but this can be removed.

Reply:
I think antlr only returns nullable in parser visitor because it's actually generating a Java abstract visitor. Programmer should really always define object hierarchy to have no nulls in kotlin

Reply:
I'm removing this file we don't use it